src/SchemaGenerator/SchemaClassGenerator.php

```diff
@@ -235,9 +248,14 @@ protected function generateEnumObject(string $objectName): bool
      */
     protected function generateArgumentsObject(string $argsObjectName, array $arguments): bool
     {
-        $objectBuilder = new ArgumentsObjectClassBuilder($this->writeDir, $argsObjectName, $this->generationNamespace);
+        if (isset($this->generatedObjects[$argsObjectName . 'ArgumentsObject'])) {
```

Comment:
Since you're doing this `$argsObjectName . 'ArgumentsObject'` twice, once for checking in the map and once for storing in it, I'd prefer if you move this concatenation result in a local variable, use it in both operations, and document why we're doing this. Maybe a link to the pull request can do, or a simple explanation.

Reply:
I wanted to do this, but then I saw that `'ArgumentsObject`' gets appended on a lot of places. I decided to remove all those places and only add it in SchemaClassGenerator. Let me know what you think :)